src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4071,6 +4438,12 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
 
             base.VisitDelegateCreationExpression(node);
             SetNotNullResult(node);
+
+            // The group is skipped by the base call. It should have a default type always
+            if (node.Argument is BoundMethodGroup group)
+            {
+                SetAnalyzedNullability(group, default);
+            }
```

Comment:
Let's replace the call to `base.VisitDelegateCreationExpression(node);` and include your change:
```
if (node.Argument is BoundMethodGroup group)
{
    VisitRvalue(group.ReceiverOpt);
    SetAnalyzedNullability(group, default);
}
else
{
    VisitRvalue(node.Argument);
}
```

Reply:
The call to `base.VisitDelegateCreationExpression(node)` above should be removed. Please add a test with a warning in the receiver to confirm we're not visiting the method group receiver twice.

---
In reply to: [265719895](https://github.com/dotnet/roslyn/pull/34017#discussion_r265719895) [](ancestors = 265719895)